backend/src/main/kotlin/be/osoc/team1/backend/controllers/StudentController.kt

```diff
@@ -91,6 +99,12 @@ class StudentController(
             .applyIf(skills.isNotEmpty()) { filterBySkills(skills) }
             .applyIf(unassignedOnly) { filterByNotYetAssigned(assignmentRepository) }
             .page(pager)
+
+        return if (view == StudentViewEnum.Full) {
+            ObjectMapper().writerWithView(StudentView.Full::class.java).writeValueAsString(filteredStudents)
+        } else {
+            ObjectMapper().writerWithView(StudentView.Basic::class.java).writeValueAsString(filteredStudents)
+        }
```

Comment:
There's quite a bit of code being repeated here. Can't you do something like this instead? (Disclaimer: I didn't test this code)
```kotlin
val view = when (view) {
    StudentViewEnum.Full -> StudentView.Full
    StudentViewEnum.Basic -> StudentView.Basic
}
return ObjectMapper().writerWithView(view::class.java).writeValueAsString(filteredStudents)
```